wrappers/wrapper-scala/src/main/scala/netsim/NetSimSocket.scala

```diff
@@ -0,0 +1,50 @@
+package netsim
+
+import helper.SimError
+import jnr.ffi.byref.NativeLongByReference
+import jnr.ffi.{NativeLong, Pointer}
+
+
+class NetSimSocket(val context: NetSimContext,
+                   socketPtr: Pointer,
+                   val simId: Long)(implicit val netSimNative: NetSimNative) {
+
+  def send(msg: NetSimMsg): SimError = {
+    //TODO how is this memory freed?
```

Comment:
you can free it as soon as you have called the `netsim_socket_send_to` function.